Dockerfile

```diff
@@ -1,104 +1,42 @@
-	&& curl -o /tmp/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.6/gosu-$(dpkg --print-architecture).asc" \
-	&& export GNUPGHOME="$(mktemp -d)" \
-	&& for server in $(shuf -e ha.pool.sks-keyservers.net \
-	                            hkp://p80.pool.sks-keyservers.net:80 \
-	                            keyserver.ubuntu.com \
-	                            hkp://keyserver.ubuntu.com:80 \
-	                            pgp.mit.edu) ; do \
-	        gpg --keyserver "$server" --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 && break || : ; \
-	    done \
-	&& gpg --batch --verify /tmp/gosu.asc /usr/local/bin/gosu \
-	&& rm -r "$GNUPGHOME" /tmp/gosu.asc \
-	&& chmod +x /usr/local/bin/gosu
-
-
-# Setup environment
-WORKDIR /var/scitran
-
-RUN mkdir -p \
-      /var/scitran/config \
-      /var/scitran/data \
-      /var/scitran/code/api \
-      /var/scitran/logs \
-      /var/scitran/keys
-
-# Declaring a volume makes the intent to map externally explicit. This enables
-# the contents to survive/persist across container versions, and easy access
-# to the contents outside the container.
-#
-# Declaring the VOLUME in the Dockerfile guarantees the contents are empty
-# for any new container that doesn't specify a volume map via 'docker run -v '
-# or similar option.
-#
-VOLUME /var/scitran/keys
-VOLUME /var/scitran/data
-VOLUME /var/scitran/logs
-
-
-# Install pip modules
-#
-# Split this out for better cache re-use.
-#
-COPY requirements.txt docker/requirements-docker.txt /var/scitran/code/api/
-
-RUN pip install --upgrade pip wheel setuptools \
-  && pip install -r /var/scitran/code/api/requirements-docker.txt \
-  && pip install -r /var/scitran/code/api/requirements.txt
-
-COPY tests /var/scitran/code/api/tests/
-RUN bash -e -x /var/scitran/code/api/tests/bin/setup-integration-tests-ubuntu.sh
-
-
-# Copy full repo
-#
-COPY . /var/scitran/code/api/
-
-COPY docker/uwsgi-entrypoint.sh /var/scitran/
-COPY docker/uwsgi-config.ini /var/scitran/config/
-COPY docker/newrelic.ini /var/scitran/config/
-
-
-
-# Inject build information into image so the source of the container can be
-# determined from within it.
-ARG BRANCH_LABEL=NULL
-ARG COMMIT_HASH=0
-COPY docker/inject_build_info.sh /
-RUN /inject_build_info.sh ${BRANCH_LABEL} ${COMMIT_HASH} \
-  && rm /inject_build_info.sh
-
-
-ENTRYPOINT ["/var/scitran/uwsgi-entrypoint.sh"]
-CMD ["uwsgi", "--ini", "/var/scitran/config/uwsgi-config.ini", "--http", "0.0.0.0:8080", "--http-keepalive", "--so-keepalive", "--add-header", "Connection: Keep-Alive" ]
+FROM python:2.7-alpine3.6 as build
+
+RUN apk add --no-cache build-base curl
+
+WORKDIR /src/nginx-unit
+
+RUN curl -L https://github.com/nginx/unit/archive/master.tar.gz | tar xz --strip-components 1
+RUN ./configure --prefix=/usr/local --modules=lib --state=/var/local/unit --pid=/var/unit.pid --log=/var/log/unit.log \
+ && ./configure python \
```

Comment:
We should check the configure options, nginx-web (or whatever) has quite a few good ones.

Reply:
On it.